docker/Dockerfile

```diff
@@ -0,0 +1,57 @@
+# https://docs.docker.com/develop/develop-images/multistage-build/
+
+#########################
+# Transpiles typescript #
+#########################
+FROM node:10-alpine as builder
+
+# node-gyp alpine deps
+RUN apk add --no-cache python make g++
```

Comment:
Wait, I thought we've already free from node-gyp now. Do you happen to know what dependency requires that?

Reply:
This is just general good practice in case we add a dep that needs it. These deps are in an intermediate build stage image, and will not be included in the final image.

Reply:
I will try removing on next pass at this.

Reply:
Ah, I see. Either removing or not sounds fine. List this [article](https://medium.com/nomic-labs-blog/turning-a-page-in-ethereum-javascript-history-4ec89136fccc) here more context of node-gyp in ethereum deps.

Reply:
Ah I see, didn't know there was a concerted effort to remove `node-gyp`. Definitely going to remove now :)